src/support/wsasocketpair.c

```diff
@@ -0,0 +1,83 @@
+// This file is a part of Julia. License is MIT: http://julialang.org/license
+
+// This is a compatibility implementation of the unix "socketpair"
+// function using the Windows Sockets 2 API.
+// Note that the return value is a WSA SOCKET (void*),
+// not a fd (int) since the msvcrt handling of
+// sockets-as-file-descriptors is generally buggy,
+// and not a Win32 HANDLE (void*) either,
+// since this is not a Win32 API call
+// (despite MSDN documentation to the contrary,
+// it will not work with ReadFile/WriteFile)
+
+#define WIN32_LEAN_AND_MEAN
+#include <winsock2.h>
+#include <io.h>
+#define AF_UNIX 1
+
+__declspec(dllexport) int wsasocketpair(int domain, int type, int protocol, SOCKET socket_vector[2]) {
```

Comment:
from our libuv fork, we have the equivalent of `pipe` that returns NamedPipe objects. this provides the ability to have the equivalent of `socketpair` instead. I suspect both would be useful to get pushed upstream into libuv.